py3status/modules/window.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+Display window properties (i.e. title, class, instance).
+
+Configuration parameters:
+    cache_timeout: refresh interval for i3-msg or swaymsg (default 0.5)
+    format: display format for this module (default "{title}")
+    hide_title: hide title on containers with window title (default False)
+    max_width: specify width to truncate title with ellipsis (default None)
+
+Format placeholders:
+    {class} window class
+    {instance} window instance
+    {title} window title
+
+Optional:
+    i3ipc: an improved python library to control i3wm and sway
+
+Examples:
+```
+# show alternative instead of empty title
+window_title {
+    format = '{title}|\u2665'
+}
+```
+
+@author shadowprince (counter), Anon1234 (async)
+@license Eclipse Public License (counter), BSD (async)
+
+SAMPLE OUTPUT
+{'full_text': u'scary snake movie - mpv'}
+
+ellipsis
+{'full_text': u'GitHub - ultrabug/py3sta…'}
+"""
+
+STRING_ERROR = "invalid ipc `{}`"
+
+
+class Ipc:
+    """
+    """
+
+    def __init__(self, parent):
+        self.parent = parent
+        self.setup(parent)
+
+    def compatibility(self, window_properties):
+        # specify width to truncate title with ellipsis
+        if self.parent.max_width:
+            title = window_properties["title"]
+            if len(title or "") > self.parent.max_width:
+                window_properties["title"] = title[: self.parent.max_width - 1] + u"…"
+
+        return window_properties
+
+
+class I3ipc(Ipc):
+    """
+    i3ipc - an improved python library to control i3wm and sway
+    """
+
+    def setup(self, parent):
+        from threading import Thread
+
+        self.parent.cache_timeout = self.parent.py3.CACHE_FOREVER
+        self.window_properties = {}
+
+        t = Thread(target=self.start)
+        t.daemon = True
+        t.start()
+
+    def start(self):
+        from i3ipc import Connection
+
+        i3 = Connection()
+        self.change_title(i3)
+        for event in ["workspace::focus", "window::close"]:
+            i3.on(event, self.clear_title)
+        for event in ["window::title", "window::focus", "binding"]:
+            i3.on(event, self.change_title)
+        i3.main()
+
+    def clear_title(self, i3, event=None):
+        self.update(i3.get_tree().find_focused())
+
+    def change_title(self, i3, event=None):
+        focused = i3.get_tree().find_focused()
+
+        # hide title on containers with window title
+        if self.parent.hide_title:
+            if (
+                focused.border == "normal"
+                or focused.type == "workspace"
+                or (
+                    focused.parent.layout in ("stacked", "tabbed")
+                    and len(focused.parent.nodes) > 1
+                )
+            ):
+                focused.window_title = None
+        self.update(focused)
+
+    def update(self, window_properties):
+        window_properties = {
+            "title": window_properties.window_title,
```

Comment:
For me with i3ipc on Sway, window_title is empty. The title is set in `name` (and window_title_async was indeed using this).

Reply:
It looks like `window_properties` is part of X11. Xwayland apps gives us this too, but not in native wayland apps... according to `swaymsg -t get_tree`. Also, `window` is set `null` there.

One had to find and test native wayland apps to reproduce your issue. :-) 

I updated #1860. We could add `{name}` and use `hide_name` to keep it straightforward because it looks like native wayland apps doesn't support window title/class/instance, but I'm not sure about having both `{title}` and `{name}`... or we could give `{title}` a value of window's name (this one is in PR right now).

Reason for first one is to keep it more honest/compatibility since `title` doesn't really exist in native wayland apps... and in case we want to add more placeholders... such as `window`, `id`, `current_border_width`, `layout`, `last_split_layout`, `border`, etc.

Reply:
I don't have any strong preference.

Does switching to `.name` as it is currently implemented won't break anything for i3? Is `{name}` defined on both WM?

Reply:
Sorry. I thought I replied hours ago.

I don't plan to use any other placeholders either. If #1860 works for you, please approve so I can merge the bugfix. Yes on `.name` working okay on i3.